projects/plugins/jetpack/modules/masterbar.php

```diff
@@ -14,6 +14,19 @@
  * @package automattic/jetpack
  */
 
-use Automattic\Jetpack\Masterbar\Main as Jetpack_Masterbar;
+/**
+ * Remove Masterbar from the old Module list.
+ * Available at wp-admin/admin.php?page=jetpack_modules
+ * We only need this function and module file until the Masterbar is fully removed from Jetpack including notices).
+ *
+ * @param array $items Array of Jetpack modules.
+ * @return array
+ */
+function remove_masterbar_module_list( $items ) {
+	if ( isset( $items['masterbar'] ) && get_option( 'wpcom_admin_interface' ) !== 'wp-admin' ) {
+		unset( $items['masterbar'] );
+	}
+	return $items;
+}
 
-Jetpack_Masterbar::init();
+add_filter( 'jetpack_modules_list_table_items', 'remove_masterbar_module_list' );
```

Comment:
The problem with this approach is that the code only runs when the module is active.
If the module is not active, it will still appear in the modules list.

We could remove the module tags from this file and from `projects/plugins/jetpack/modules/module-headings.php`, then the module will disappear by itself.
But that might interfere with the module functionality.

Instead, we could place this code in the `class-deprecate.php`, and then remove it along with the module.
Do you think that would work?